mm2src/mm2_tests.rs

```diff
@@ -8312,3 +8329,216 @@ fn test_get_orderbook_with_same_orderbook_ticker() {
+        "userpass": mm_bob.userpass,
+        "method": "setprice",
+        "base": "RICK",
+        "rel": "MORTY",
+        "price": 0.9,
+        "volume": "0.9",
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!setprice: {}", rc.1);
+
+    let rc = block_on(mm_bob.rpc(json! ({
+        "userpass": mm_bob.userpass,
+        "mmrpc": "2.0",
+        "method": "get_public_key",
+        "params": {},
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!get_public_key: {}", rc.1);
+    log!("GetPublicKeyResult "[rc.1]);
+    let get_public_key_res: Json = json::from_str(&rc.1).unwrap();
+    let bob_pubkey = get_public_key_res["result"]["public_key"].as_str().unwrap();
+
+    // Alice coins don't have required_confirmations and requires_notarization set
+    let alice_coins = json!([
+        {"coin":"RICK","asset":"RICK","rpcport":8923,"txversion":4,"overwintered":1,"protocol":{"type":"UTXO"}},
+        {"coin":"MORTY","asset":"MORTY","rpcport":11608,"txversion":4,"overwintered":1,"protocol":{"type":"UTXO"}},
+    ]);
+
+    let mut mm_alice = MarketMakerIt::start(
+        json!({
+            "gui": "nogui",
+            "netid": 9998,
+            "passphrase": "alice passphrase",
+            "rpc_password": "password",
+            "coins": alice_coins,
+            "seednodes": [fomat!((mm_bob.ip))],
+        }),
+        "password".into(),
+        None,
+    )
+    .unwrap();
+    let (_dump_log, _dump_dashboard) = mm_alice.mm_dump();
+    log!({"Alice log path: {}", mm_alice.log_path.display()});
+
+    log! ({"enable_coins (alice): {:?}", block_on (enable_coins_rick_morty_electrum(&mm_alice))});
+
+    // setting the price will trigger Alice's subscription to the orderbook topic
+    // but won't request the actual orderbook
+    let rc = block_on(mm_alice.rpc(json! ({
+        "userpass": mm_alice.userpass,
+        "method": "setprice",
+        "base": "RICK",
+        "rel": "MORTY",
+        "price": "1",
+        "volume": "0.1",
+        "cancel_previous": false,
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!setprice: {}", rc.1);
+
+    block_on(
+        mm_alice.wait_for_log((MIN_ORDER_KEEP_ALIVE_INTERVAL * 2) as f64, |log| {
+            log.contains(&format!("Inserting order OrderbookItem {{ pubkey: \"{}\"", bob_pubkey))
+        }),
+    )
+    .unwrap();
+
+    log!("Get RICK/MORTY orderbook on Alice side");
+    let rc = block_on(mm_alice.rpc(json! ({
+        "userpass": mm_alice.userpass,
+        "method": "orderbook",
+        "base": "RICK",
+        "rel": "MORTY",
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!orderbook: {}", rc.1);
+
+    let alice_orderbook: OrderbookResponse = json::from_str(&rc.1).unwrap();
+    log!("Alice orderbook "[alice_orderbook]);
+    assert_eq!(
+        alice_orderbook.asks.len(),
+        2,
+        "Alice RICK/MORTY orderbook must have exactly 2 ask"
+    );
+    let bob_orders_in_orderbook = alice_orderbook
```

Comment:
`iter().find(|entry| entry.pubkey == bob_pubkey).unwrap()` would be sufficient.

Reply:
Done